tests/scripts/check_abi_check.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python3
+
+""" Check abi_check.py is working correctly
+"""
+
+# Copyright The Mbed TLS Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+import traceback
+import sys
+import os
+import subprocess
+import re
+
+SCRIPT_PATH = "scripts/abi_check.py"
+
+def run_abi_check(old_tag, new_tag):
+    return subprocess.check_output([SCRIPT_PATH, "-o", old_tag, "-n", new_tag],
+                                   stderr=subprocess.STDOUT).decode("utf-8")
+
+def pattern_create(lines):
+    return "^["+ "".join(["("+line+")" for line in lines]) +"]$"
+
+class TestAbiCheck(unittest.TestCase):
+    ''' This class test abi check script
+    '''
+
+    @staticmethod
+    def check_repo_path():
+        if not all(os.path.isdir(d) for d in ["include", "library", "tests"]):
+            raise Exception("Must be run from Mbed TLS root")
+
+    def test_abi_same(self):
+        ''' This test checks no false positive
+        '''
+        possible_lines = ["Checking.*", "No compatibility issues.*", "PASS.*", "Info.*"]
+        pattern = pattern_create(possible_lines)
+        out = run_abi_check("mbedtls-3.2.0","mbedtls-3.2.1")
+        self.assertIsNone(re.fullmatch(pattern, out, re.MULTILINE))
+
+    def test_abi_different(self):
```

Comment:
`abi_check` does several things, currently 4: API differences, ABI differences, generated file differences, storage format test differences. We should have a negative test for each of them: e.g. there should at least one test case with (compatible ABI, incompatible ABI, compatible generated files, compatible storage format tests), and one with (compatible API, compatible ABI, compatible generated files, incompatible storage format tests), and one with (compatible API, compatible ABI, incompatible generated files, compatible storage format tests) (I'm not sure if incompatible API, compatible ABI is possible with abi_check's definition of API/ABI compatibility).

The task description for #5409 specifically requests negative testing for API/ABI compatibility, and there's a separate task #5410 for generated test cases. So for #5409 it's enough to focus on API/ABI differences, but the test has to fail if `abi_check.py` only reports incompatibilities in generated tests.

Reply:
`_read_storage_tests` records test cases if they either come from an automatically generated file or the function name contains `read`. So for #5410, there should be a test case that removes a `read` test from a non-automatically-generated file, and a test case that removes a non-`read` test from an automatically generated file.

c4b814a9c2c192f8f52908645720ced9cb817447 is an ABI break and not an API break. All enum identifiers still exist afterwards, so source programs will still work, so it's a compatible API. But the values of some enum identifiers change, so some compiled programs won't work anymore, so it's an ABI break.